server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -310,8 +322,11 @@ onStart serverEnv wsConn (StartMsg opId q) = catchAndIgnore $ do
       logOpEv ODStarted (Just reqId)
       -- log the generated SQL and the graphql query
       L.unLogger logger $ QueryLog query genSql reqId
+      t1 <- liftIO TC.getCurrentTime -- for measuring response time purposes
       resp <- liftIO $ runExceptT action
-      either (postExecErr reqId) sendSuccResp resp
+      t2 <- liftIO TC.getCurrentTime -- for measuring response time purposes
```

Comment:
It might be worth extracting this pattern into a helper function with a type like this:

```haskell
withElapsedTime :: MonadIO m => m a -> m (DiffTime, a)
```